### ISSUE TITLE
Double bridging with MOI layers

### ISSUE THREAD

**Issue by blegat** [jump-dev/MathOptInterface.jl#2379]
Convex.Optimizer add bridges to its inner optimizer because bridges might be needed to transformed the conic problem it generates. Moreover, `Convex.Optimizer` returns `true` for `supports_incremental_interface` since otherwise `default_copy_to` would error:
https://github.com/jump-dev/MathOptInterface.jl/blob/ede6d1cb7a3e233de5cd39a04ca888da4acd2163/src/Utilities/copy.jl#L479-L482
Because of this, after `MOI.instantiate`, we get `MOI.LazyBridgeOptimizer(Convex.Optimizer(MOI.LazyBridgeOptimizer(Clarabel.Optimizer)))` and this creates issues because bridges use negative indices to distinguish its indices but if we have two layers of bridges without a caching optimizer in between that's not working.
A workaround would be for `Convex.Optimizer` to return `false` for `default_copy_to` but the error in `default_copy_to` prevents this.
I think the issue is kind of like https://github.com/jump-dev/MathOptInterface.jl/issues/2361 where these trait should really just be "does `instantiate` need to add a cache layer if it adds a bridge layer ?" and if we remove these error then it can be just that.

This is a recurring issue with MOI layers. For instance, DiffOpt also adds an internal bridge layers and uses `default_copy_to` but it does it by directly copying to the internal optimizer thus avoiding the need to define `supports_incremental_interface` even if it should be true. I expect that ParametricOptInterface should have the same issue.

**Comment by odow** [comment]
Do you have a reproducible example of this failing?

**Comment by blegat** [comment]
```julia
import MathOptInterface as MOI
import Convex
import ECOS
model = MOI.instantiate(() -> Convex.Optimizer(ECOS.Optimizer), with_bridge_type=Float64)
MOI.Test.test_add_parameter(model, MOI.Test.Config())
```
Throws
```
The index MathOptInterface.ConstraintIndex{MathOptInterface.VariableIndex, MathOptInterface.Parameter{Float64}}(-1) is invalid. Note that an index becomes invalid after it has been deleted.
```
but that far from the only test failing, it's just the first one. The issue is that the bridge layer things that it's his constraints since it has a negative index but it's one of the inner bridging layer.

**Comment by odow** [comment]
Thanks. I was just trying various things in JuMP without success.

One fix is for Convex to maintain an `index_map` so that it does not blindly pass indices from the inner solver to the outer layer. I don't know if there is an easy fix for this in MOI.

**Comment by blegat** [comment]
Yes, Miles suggested during the meeting to simply add an inner caching optimizer layer in Convex.jl. That's a lot of caches but it's hopefully temporary

**Comment by odow** [comment]
I don't think it needs an inner cache. Convex just needs to not return negative indices.

**Comment by blegat** [comment]
Convex is not returning negative indices but the bridge layer used by inner solver might

**Comment by odow** [comment]
Yeah, it just returns the index from the inner. But if it had an additional layer of indirection, it could map 1-based outer indices  to the inner index.

https://github.com/jump-dev/Convex.jl/blob/4424337bfca6a1c2dac61e74e50e2c7487dc7d46/src/MOI_wrapper.jl#L91-L94

**Comment by blegat** [comment]
Yes, but I'd rather add a layer of CachingOptimizer that already does that than copy-paste part of CachingOptimizer's code into Convex.jl

**Comment by odow** [comment]
So should we move this issue to Convex.jl? It doesn't seem like something that should be fixed in MOI.
